scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SchedulingDSLCompilationService.java

```diff
@@ -24,7 +24,7 @@ public SchedulingDSLCompilationService(final TypescriptCodeGenerationService typ
     this.typescriptCodeGenerationService = typescriptCodeGenerationService;
     final var schedulingDslCompilerRoot = System.getenv("SCHEDULING_DSL_COMPILER_ROOT");
     final var schedulingDslCompilerCommand = System.getenv("SCHEDULING_DSL_COMPILER_COMMAND");
-    this.nodeProcess = new ProcessBuilder("node", schedulingDslCompilerCommand)
+    this.nodeProcess = new ProcessBuilder("node", "--experimental-vm-modules", schedulingDslCompilerCommand)
```

Comment:
Do we know the absolute `node` path to use here?

Reply:
It's different depending on whether you're running unit tests, or running in Docker 🤔 In Docker it's wherever nvm installs it to. In unit testing, it's in the gradle directory.

I could potentially make a `NODE_PATH` environment variable, but I'm not sure if that's better from a security perspective.

Open to ideas for how to do this better

Reply:
hmm yeah `NODE_PATH` is decent but you're right it potentially moves the issue to another env var. I think for now `NODE_PATH` is a good compromise because `PATH` is system-wide and `NODE_PATH` is set for the application specifically.

Reply:
I've gone the `NODE_PATH` route :+1:

Reply:
Alas, I cannot for the life of me figure out what the absolute path to node is in Github Actions

Reply:
Alright - a compromise I've reached is that in production, `NODE_PATH` will be set to `"/root/.nvm/versions/node/v${NODE_VERSION}/bin/node"`. In the `./gradlew test` environment, it will look in the `.gradle` directory to try to find the version of node that gradle installed. If it can't find that, it will set `NODE_PATH` to `"node"`, and fall back to whatever node is on the path.

I think this follows the spirit of the rule, since in production, it will always use an absolute path. I think it's okay to bend the rule in the unit testing environment (though I am still confused why gradle in the github action doesn't install node in the same location as when I run it on my machine 😕 )